jujugui/static/gui/src/app/init.js

```diff
@@ -659,13 +693,97 @@ class GUIApp {
     @param {Function} next - Run the next route handler, if any.
   */
   _handleModelState(state, next) {
-    const modelAPI = this.modelAPI;
-    if (this.modelUUID !== state.model.uuid ||
-        (!modelAPI.get('connected') && !modelAPI.get('connecting') &&
-        !modelAPI.loading)) {
-      this._switchModelToUUID(state.model.uuid);
+    const newModelUUID = state.model.uuid;
+    if (this.activeModelUUID === newModelUUID) {
+      if (this.modelConnectionStatus === 'ready') {
+        next();
+        return;
+      } else if (this.modelConnectionStatus === 'connecting') {
+        // If we get in here it's because the application has dispatched multiple
+        // times but we do not want to connect multiple times.
+        return;
+      }
     }
-    next();
+
+    const modelURL = jujulib.generateModelURL(
+      this.applicationConfig.apiAddress, newModelUUID);
+
+    const connectionOptions = {
+      debug: true,
+      facades: [
+        // Sort facades alphabetically.
+        require('@canonical/jujulib/api/facades/application-v6.js'),
```

Comment:
This is going to do a require every time we change models. There's nothing dynamic going on here so would it be better to store this array at the top of the file with the other requires?

Reply:
Yes good point, it should cache, but I'll move it out to be explicit.

Reply:
done